arwen/elrondapi/elrondei.go

```diff
@@ -377,9 +383,56 @@ func transferValue(context unsafe.Pointer, gasLimit int64, destOffset int32, val
 	metering.UseGas(gasToUse)
 
 	invBytes := arwen.InverseBytes(value)
-	invValue := big.NewInt(0).SetBytes(invBytes)
-	boundGasLimit := metering.BoundGasLimit(gasLimit)
-	output.Transfer(dest, send, boundGasLimit, invValue, data)
+
+	output.Transfer(dest, send, 0, big.NewInt(0).SetBytes(invBytes), data)
+
+	return 0
+}
+
+//export asyncCall
+func asyncCall(context unsafe.Pointer, destOffset int32, valueOffset int32, dataOffset int32, length int32) int32 {
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+	output := arwen.GetOutputContext(context)
+
+	send := runtime.GetSCAddress()
+	dest, err := runtime.MemLoad(destOffset, arwen.AddressLen)
+	if withFault(err, context) {
+		return 1
```

Comment:
instead of return SetRuntimeBreakPointValue.

Reply:
Fixed: de-duplicated all definitions of `withFault`, which is now found in `helpers.go`, and `WithFault` always calls `runtime.SignalUserError()` which sets the breakpoint.